### PR TITLE
Cinder snapshots

### PR DESCRIPTION
**Pull request by j-griffith**
Implements Cinder snapshot create/delete/rollback for Cinder OpenStack cloudprovider.

### CODE REVIEW

File: pkg/cloudprovider/providers/openstack/metadata.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+		return nil, err
+	}
+	defer f.Close()
+
+	return parseMetadata(f)
+}
+
+func getMetadataFromMetadataService() (*Metadata, error) {
+	// Try to get JSON from metdata server.
+<<<<<<< HEAD
```
Comment (rootfs) [inline pkg/cloudprovider/providers/openstack/metadata.go]: merge conflict

File: pkg/cloudprovider/providers/openstack/metadata.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
Comment (rootfs) [inline pkg/cloudprovider/providers/openstack/metadata.go]: 2017

File: pkg/volume/cinder/processor.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+			VolumeID:  volumeID,
+			FSType:    "ext4",
+			Partition: 0,
+			ReadOnly:  false,
+		},
+	}
+	return pv, nil, nil
+}
+
+// Describe an EBS volume snapshot status for create or delete.
```
Comment (rootfs) [inline pkg/volume/cinder/processor.go]: EBS -> Cinder

**Comment by rootfs** [comment]
most looks good to me.

can you also update https://github.com/rootfs/snapshot/blob/master/cmd/snapshot-controller/snapshot-controller.go and https://github.com/rootfs/snapshot/blob/master/cmd/snapshot-pv-provisioner/snapshot-pv-provisioner.go?

File: pkg/cloudprovider/providers/openstack/metadata.go (hunk 1 above)
Comment (j-griffith) [inline pkg/cloudprovider/providers/openstack/metadata.go]: Bahh!  Yeah, and looking through this there's a few things broken on that merge.  Should be easy to update/fix.

File: pkg/volume/cinder/processor.go (hunk 3 above)
Comment (j-griffith) [inline pkg/volume/cinder/processor.go]: :)

**Comment by j-griffith** [comment]
I've added the controller pieces, but noticed a few other things that I really should clean up.  I'll get these turned around hopefully tomorrow or Friday.  At a conference but should be able to find some time to work on this inbetween sessions.  Thanks!!

**Comment by rootfs** [comment]
looking forward!

**Pull request merged**